dallinger/frontend/templates/dashboard_mturk.html

```diff
@@ -1,31 +1,167 @@
-    <p>Last updated {{ data["last_updated"] }}
-        <button type="button" class="btn btn-secondary" onclick="location.reload()">Reload</button>
-    </p>
 
+    <h2>Current HIT Information</h2>
 
     {% if data["hit_info"] %}
         <table class="table table-sm table-striped">
         {% for key, value in data["hit_info"].items() %}
             <tr><th>{{ key }}</th><td>{{ value }}</td></tr>
+
         {% endfor %}
         </table>
+
+        <h3>Preview of AD page running <a href="{{ data['ad_url'] }}">on Heroku</a></h3>
+        <iframe width="100%" height="600" src="{{  data['ad_url'] }}" /></iframe>
     {% else %}
         <div class="alert alert-warning">
             HIT data not available until first participant joins.
         </div>
     {% endif %}
 
+
 {% endif %}
 
+<!-- HIT Extension Modal -->
+<div class="modal fade"
+     id="hit-extension-modal"
+     data-backdrop="static"
+     data-keyboard="false"
+     tabindex="-1"
+     role="dialog"
+     aria-labelledby="hit-extension-form-label"
+     aria-hidden="true">
+  <div class="modal-dialog">
+    <div class="modal-content">
+      <div class="modal-header">
+        <h5 class="modal-title" id="hit-extension-form-label">Update MTurk HIT Expiration and Assignments</h5>
+        <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+          <span aria-hidden="true">&times;</span>
+        </button>
+      </div>
+      <div class="modal-body">
+        <form id="hit-extension-form">
+          <div class="form-group">
+            <input type="hidden"
+                   id="original-expiration"
+                   name="original-expiration"
+                   value="{{ data['hit_expiration'] }}" />
+            <label for="new-expiration">Extend until:</label>
+            <input type="datetime-local"
+                   id="new-expiration"
+                   class="form-control"
+                   value="{{ data['hit_expiration'] }}"
+                   name="new-expiration">
+
+            <label for="additional-assignments">Additional assignments:</label>
+            <input type="number"
+                   id="additional-assignments"
+                   class="form-control"
+                   value="0"
+                   name="additional-assignments">
+
+          </div>
+          <button type="button" id="extend-hit-cancel" class="btn btn-secondary" data-dismiss="modal">Cancel</button>
+          <button type="button" id="extend-hit-button" class="btn btn-primary">Copy CLI command to clipboard</button>
+          <div id="extension-result" class="alert alert-success invisible"><strong>Dallinger CLI command copied!</strong>
+            <div id="extension-command"></div>
+          </div>
+        </form>
+      </div>
+    </div>
+  </div>
+</div>
+
+{% endblock %}
+
+{% block scripts %}
+<script type="text/javascript">
+    {% if data["hit_info"] %}
+    const HITId = "{{ data['hit_info']['HIT Id'] }}";
```

Comment:
Does this stuff need to be in the global/window namespace?

Reply:
No, definitely not. I'll move it to function scope.